### PR TITLE
#12: Added support for value tables. Adapted dependent code.

### PR DESCRIPTION
**Pull request by redcatbear**

### CODE REVIEW

File: src/main/java/com/exasol/sql/dml/rendering/InsertRenderer.java [hunk 1]
```diff
@@ -84,4 +95,16 @@ public static InsertRenderer create() {
...
+    public void visit(final InsertValues insertValues) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void leave(final InsertValues insertValues) {
+        // TODO Auto-generated method stub
+
+    }
```
Comment (morazow) [inline src/main/java/com/exasol/sql/dml/rendering/InsertRenderer.java]: Are they supposed to be empty?

File: src/main/java/com/exasol/sql/dql/FromClause.java [hunk 2]
```diff
@@ -22,28 +23,39 @@ public FromClause(final Fragment root) {
...
     public FromClause tableAs(final String name, final String as) {
         this.tables.add(new Table(getRoot(), name, as));
         return this;
     }
 
+    /**
+     * Create a {@link FromClause} from a value table
+     *
+     * @param valueTable table of value expressions
+     * @return new instance
```
Comment (morazow) [inline src/main/java/com/exasol/sql/dql/FromClause.java]: Returns `FROM clause` here also.

File: src/main/java/com/exasol/sql/dql/ValueTable.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    private ValueTableRow.Builder createLastRowBuilder() {
+        final ValueTableRow.Builder builder = ValueTableRow.builder(this.root);
+        if (!isEmpty()) {
+            builder.add(getLastRow().getExpressions());
+        }
+        return builder;
+    }
+
+    private synchronized void ammendLastRow(final ValueTableRow row) {
```
Comment (morazow) [inline src/main/java/com/exasol/sql/dql/ValueTable.java]: Change `ammend` to `amend`.

File: src/main/java/com/exasol/sql/expression/BooleanLiteral.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    public static BooleanExpression of(final boolean value) {
+        return new BooleanLiteral(value);
+    }
+
+    /**
+     * Create a new {@link BooleanLiteral} instance from a String
+     *
+     * @param value the string to be turned into a literal
+     * @return new Literal instance
```
Comment (morazow) [inline src/main/java/com/exasol/sql/expression/BooleanLiteral.java]: `Literal` to `BooleanLiteral`

File: pom.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.exasol</groupId>
   <artifactId>sql-statement-builder</artifactId>
-  <version>0.3.0</version>
+  <version>0.2.0</version>
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply pom.xml]: We have only one release so far (0.1.0), so 0.3.0 was the typo.

[the remark this answers is not in the document]
Comment (redcatbear) [reply pom.xml]: Correction. I found a version 0.2.0 on Maven Central, so we are back to 0.3.0.

File: src/main/java/com/exasol/sql/dml/rendering/InsertRenderer.java (hunk 1 above)
Comment (redcatbear) [reply src/main/java/com/exasol/sql/dml/rendering/InsertRenderer.java]: Removed dead code.

File: src/main/java/com/exasol/sql/dql/FromClause.java (hunk 2 above)
Comment (redcatbear) [reply src/main/java/com/exasol/sql/dql/FromClause.java]: Fixed.

File: src/main/java/com/exasol/sql/dql/ValueTable.java (hunk 3 above)
Comment (redcatbear) [reply src/main/java/com/exasol/sql/dql/ValueTable.java]: Fixed.

File: src/main/java/com/exasol/sql/expression/BooleanLiteral.java (hunk 4 above)
Comment (redcatbear) [reply src/main/java/com/exasol/sql/expression/BooleanLiteral.java]: Fixed.

**Comment by morazow** [comment]
Looks good!

**Pull request merged**